tmt/utils.py

```diff
@@ -955,6 +902,9 @@ def _check_or_create_workdir_root_with_perms() -> None:
 
             # Create the workdir
             create_directory(workdir, 'workdir', quiet=True)
+
+        # print(f'{self.__class__.__name__}._workdir_init(id_={id_}) => {workdir}')
+        self._logger.add_logfile_handler(os.path.join(workdir, tmt.log.LOG_FILENAME))
```

Comment:
Does this has to happen? When I provide my own logger I don't expect that common will add handlers there.

Reply:
Added a comment in 7f0b5ba7ad99741430d476360a5f8ef74901bf96 - it probably does have to happen, and IMHO the fix should involve the refactoring of how workdir is created, this nasty logfile bit is just a consequence of lazy workdir creation.

Reply:
Thanks for providing a detailed explanation. I guess that's all we can do here now.